src/main/java/com/pubnub/api/managers/RetrofitManager.java

```diff
@@ -104,6 +107,20 @@ public RetrofitManager(PubNub pubNubInstance) {
                             this.pubnub.getConfiguration().getConnectTimeout()
                     ).retryOnConnectionFailure(false)
             );
+
+            this.pubnub.addListener(new SubscribeCallback.BaseSubscribeCallback() {
+                @Override
```

Comment:
I understand that this solution needs to be hacky, so IMO we need to have a detailed description for that change in merge commit.

Reply:
Yes. I'm going to add comment in the code as well